### PR TITLE
Refactor `__eq__` and `__hash__` methods

### PR DESCRIPTION
**Pull request by CunliangGeng**
To solve #136.

The `ObjectLink` is not changed due to lack of understanding of this class. It'll be refactored when we start refactoring the scoring code.

### CODE REVIEW

File: src/nplinker/genomics/gcf.py [hunk 1]
```diff
@@ -13,7 +12,7 @@
 
 class GCF():
 
-    def __init__(self, gcf_id: str) -> None:
+    def __init__(self, gcf_id: str, /) -> None:
```
Comment (hechth) [inline src/nplinker/genomics/gcf.py:15]: Can you explain what this does?  the `/` in the arguments?

**Review by hechth — APPROVED** [review]

File: src/nplinker/genomics/gcf.py (hunk 1 above)
Comment (CunliangGeng) [reply src/nplinker/genomics/gcf.py:15]: Parameters before “/” are positional-only parameters (see [PEP570](https://peps.python.org/pep-0570/#positional-only-parameters)). 
In this case, users must use `gcf = GCF("f001")` and cannot use `gcf = GCF(gcf_id="f001")`. So later we can easily rename `gcf_id` without bothering users.

The "/" is also used when the name of parameter does not really matter:

image

File: tests/genomics/test_mibig_loader.py [hunk 2]
```diff
@@ -79,4 +79,4 @@ def test_parse_bgc_metadata_json():
     bgc = parse_bgc_metadata_json(str(json_file))
     assert isinstance(bgc, BGC)
     assert bgc.bgc_id == "BGC0000001"
-    assert bgc.mibig_bgc_class == ["Polyketide"]
+    assert bgc.mibig_bgc_class == ("Polyketide",)
```
Comment (gcroci2) [inline tests/genomics/test_mibig_loader.py:82]: Why do we need the comma here? (same in tests/genomics/test_mibig_metadata.py)

**Review by gcroci2 — APPROVED** [review]

File: tests/genomics/test_mibig_loader.py (hunk 2 above)
Comment (CunliangGeng) [reply tests/genomics/test_mibig_loader.py:82]: `("sth")` is same as as string `"sth"`. Giving a comma, you will get a tuple not string.

**Pull request merged**